### DISCUSSION TITLE
Set definite max and min viewport co ordinates

### DISCUSSION THREAD

**Discussion by navaniharsh31** [xyflow/xyflow#3542] [Q&A] [answered]
How can i set a definite x and y viewport coordinates at start. If this gets filled it should expands. And also how can i avoid nodes getting a negative x or y coordinate?

**Comment by bcakmakoglu** [comment, marked as answer]
You can limit the area of the viewport by using [`translateExtent`.](https://reactflow.dev/docs/api/react-flow-props/#translateextent)
As for nodes, you can use [`nodeExtent`](https://reactflow.dev/docs/api/react-flow-props/#translateextent)

**Reply by bcakmakoglu** [reply]
For example if you want to limit panning the viewport from 0,0 to 1000,1000 you could use

```ts
translateExtent={[[0,0],[1000,1000]]}
```

It works the same for `nodeExtent`
